webapp/tests/forms/flows/test_unlock_code_revocation_flow.py

```diff
@@ -179,8 +179,8 @@ def test_if_user_exists_and_dob_correct_and_unlock_code_revocation_got_through_t
 
     def test_if_user_exists_and_dob_correct_and_unlock_code_revocation_got_through_then_user_is_deleted(self):
         existing_idnr = '04452397687'
-        correct_dob = '01.01.1985'
-        create_user(existing_idnr, correct_dob, '0000')
+        correct_dob = ['1', '1', '1985']
```

Comment:
Why can't we keep the data as a string? Didn't we do it like that for the idnr?

Reply:
Mh, no. This is the input as given by the form. And the form - in both cases, idnr and dob, sends the data as a list of strings. Only difference is (and that is why I did not have to change this test for idnr) that the validator of date relies on a special representation (the list), while the idnr validator can handle both string and list.

I'd be hesitant to store the date as string and here's why:
- for IdNr we have two representations: `list` if not (yet) valid or in the form, `string` if valid or in the stored_data
- for date we'd have three represenations: `list` if in the form, `string` if not (yet) valid and `date` if valid or in the stored_data
Do you disagree?

Reply:
Thanks, everything makes sense now